### PR TITLE
Add config options to client + tests (fixes #6)

### LINKED ISSUE #6: Make payments client more configurable
* Allow payments host to be defined (with default on the dev server pay.dev)
* Have httpsOnly - defaults to true and throws if `config.paymentHost` is not https - meaning consumer has to explicitly override the setting.
* Only allow http/https for the iframeSrc
* Warn if httpsOnly has been set to false.
* Require accessToken + product strings

### PR DESCRIPTION
**Pull request by muffinresearch**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+module.exports = {
+  getProtocol: function(url) {
+    var link = document.createElement('a');
+    link.href = url;
+    return link.protocol;
```
Comment (andymckay) [inline src/utils.js]: can't you just do a regex on the url?

File: tests/test.client.js [hunk 2]
```diff
@@ -3,13 +3,56 @@
...
+
+  it('should throw if missing an accessToken name', function() {
+    assert.throw(function() {
+      /*eslint-disable no-new */
+      new PaymentsClient({
+        httpsOnly: false,
+        product: 'whatever',
+      });
+    }, Error, /An 'accessToken' string must/);
+  });
```
Comment (andymckay) [inline tests/test.client.js]: nice tests

File: tests/test.client.js [hunk 3]
```diff
@@ -65,4 +108,13 @@ describe('Test client', function() {
     }, 0);
   });
 
+  it('iframeSrc should be have product and accessToken', function() {
+    this.client.show();
+    var iframeNode = this.modalParent.querySelector('iframe');
+    assert.notEqual(iframeNode, null);
+    assert.include(iframeNode.src, 'http://pay.dev:8000/?');
+    assert.include(iframeNode.src, 'product=something-awesome');
+    assert.include(iframeNode.src, 'access_token=blah-blah-acess-token-blah');
```
Comment (andymckay) [inline tests/test.client.js]: hehe typo in access, which doesn't matter in the slightest, but it stood out :)

File: tests/test.utils.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var utils = require('utils');
+
+
+describe('utils.getProtocol()', function() {
+
+  it('can parse javsscript psuedo protocols', function() {
```
Comment (andymckay) [inline tests/test.utils.js]: typo in javascript

File: tests/test.utils.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+describe('utils.serialize()', function() {
+
+  it('should serialize object', function() {
+    assert.equal(utils.serialize({foo: 'bar', baz: 'zup'}),
+                 'foo=bar&baz=zup');
+  });
+
+  it('should urlencode keys + values', function() {
+    assert.equal(
+      utils.serialize({'album name': 'Back in Black', 'artist': 'AC/DC'}),
```
Comment (andymckay) [inline tests/test.utils.js]: i approve of this album

**Comment by andymckay** [comment]
r+wc

File: src/utils.js (hunk 1 above)
Comment (muffinresearch) [inline src/utils.js]: Yeah, could do - though was thinking ahead we're probably going to want to extend this to get hosts later.

**Pull request merged**